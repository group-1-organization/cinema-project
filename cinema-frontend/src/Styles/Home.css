.logo-header {
    padding: 15px;
    text-align: center;
    background: #0e254d;
    
}

.text-colour {
    color: black;
}

