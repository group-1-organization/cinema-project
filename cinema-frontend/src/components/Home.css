.header {
    padding: 15px;
    text-align: center top;
    background: #0e254d;
    
}

.text-colour {
    color: black;
}

