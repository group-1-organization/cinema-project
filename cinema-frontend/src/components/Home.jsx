import Container from 'react-bootstrap/Container';
import { Carousel } from 'react-bootstrap';
import { Card } from 'react-bootstrap';
import { CardDeck } from 'react-bootstrap';
import '../Styles/Home.css'

const Home = () => {
    return (
        <div>
            <div class="logo-header">
                <img src="/images/BannerResize.png"
                    alt="QA Cinema banner"
                    class="logo-center" />
            </div>
            <br />
            <Container>
                <Carousel>
                    <Carousel.Item interval={2000}>
                        <img
                            className="d-block w-100"
                            src="/images/AQP2.jpg"
                            alt="A Quiet Place 2 banner"
                        />
                        <Carousel.Caption>
                        </Carousel.Caption>
                    </Carousel.Item>
                    <Carousel.Item interval={2000}>
                        <img
                            className="d-block w-100"
                            src="/images/ITH2.png"
                            alt="In The Heights banner"
                        />

                        <Carousel.Caption>
                        </Carousel.Caption>
                    </Carousel.Item>
                    <Carousel.Item interval={2000}>
                        <img
                            className="d-block w-100"
                            src="/images/BW.jpg"
                            alt="Black Widow banner"
                        />

                        <Carousel.Caption>
                        </Carousel.Caption>
                    </Carousel.Item>
                </Carousel>
            </Container>
            <br />
            <div class="text-colour">
                <Container>
                    <CardDeck>
                        <Card>
                            <Card.Img variant="top" src="/images/premiumdecor.jpg" width="348" height="220" />
                            <Card.Body>
                                <Card.Title>QA Cinema</Card.Title>
                                <Card.Text>
                                    QA Cinema provides you with the latest, biggest films at afforable prices; With comfortable seating and reasonably priced food & drink to boot, what reason is there not to come spend an evening with us? Here at QACinema.com you can browse the latest films, those still to come, book tickets and find out more about us.
                                </Card.Text>
                            </Card.Body>
                        </Card>
                        <Card>
                            <Card.Img variant="top" src="/images/hire.jpeg" width="348" height="220" />
                            <Card.Body>
                                <Card.Title>Private hire!</Card.Title>
                                <Card.Text>
                                    Bothered by all the strange plebians in the cinema? Want to place some games on that big screen instead? Then consider hiring one of our screens privately for you and your friends, with prices starting from just Â£100!
                                </Card.Text>
                            </Card.Body>
                        </Card>
                        <Card>
                            <Card.Img variant="top" src="/images/safety.jpg" width="348" height="220" />
                            <Card.Body>
                                <Card.Title>Covid safety guidelines</Card.Title>
                                <Card.Text>
                                    As a result of the pandemic, we have set safety measures in place to help keep our customers safe, including numerous sanitising stations and socially distanced seating.
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </CardDeck>
                </Container>
            </div>
        </div>
    )
}

export default Home
