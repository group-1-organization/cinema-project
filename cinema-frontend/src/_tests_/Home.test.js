import renderer, { act } from 'react-test-renderer';
import Home from '../components/Home';

test('Homepage matches snapshot', () => {
    const tree = renderer.create(<Home />);

    expect(tree).toMatchSnapshot();
});