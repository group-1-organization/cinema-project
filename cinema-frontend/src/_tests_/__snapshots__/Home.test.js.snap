// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<div>
  <div
    class="logo-header"
  >
    <img
      alt="QA Cinema banner"
      class="logo-center"
      className="img-fluid"
      src="/images/BannerResize.png"
    />
  </div>
  <div
    className="carousel slide"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="active carousel-item"
        interval={2000}
      >
        <img
          alt="A Quiet Place 2 banner"
          className="d-block w-100"
          src="/images/f9-banner.jpg"
        />
        <div
          className="carousel-caption"
        />
      </div>
      <div
        className="carousel-item"
        interval={2000}
      >
        <img
          alt="In The Heights banner"
          className="d-block w-100"
          src="/images/bw-banner.jpg"
        />
        <div
          className="carousel-caption"
        />
      </div>
      <div
        className="carousel-item"
        interval={2000}
      >
        <img
          alt="Black Widow banner"
          className="d-block w-100"
          src="/images/sj-banner.jpg"
        />
        <div
          className="carousel-caption"
        />
      </div>
    </div>
    <a
      className="carousel-control-prev"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="d-flex align-items-stretch col-lg-4 col-md-6"
      >
        <div
          className="mt-4 card bg-dark text-light"
        >
          <img
            className="card-img-top"
            height="220"
            src="/images/premiumdecor.jpg"
            width="348"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              QA Cinema
            </div>
            <p
              className="card-text"
            >
              QA Cinema provides you with the latest, biggest films at afforable prices; With comfortable seating and reasonably priced food & drink to boot, what reason is there not to come spend an evening with us? Here at QACinema.com you can browse the latest films, those still to come, book tickets and find out more about us.
            </p>
          </div>
        </div>
      </div>
      <div
        className="d-flex align-items-stretch col-lg-4 col-md-6"
      >
        <div
          className="mt-4 card bg-dark text-light"
        >
          <img
            className="card-img-top"
            height="220"
            src="/images/hire.jpeg"
            width="348"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Private hire!
            </div>
            <p
              className="card-text"
            >
              Bothered by all the strange plebians in the cinema? Want to place some games on that big screen instead? Then consider hiring one of our screens privately for you and your friends, with prices starting from just Â£100!
            </p>
          </div>
        </div>
      </div>
      <div
        className="d-flex align-items-stretch col-lg-4 col-md-6"
      >
        <div
          className="mt-4 card bg-dark text-light"
        >
          <img
            className="card-img-top"
            height="220"
            src="/images/safety.jpg"
            width="348"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Covid safety guidelines
            </div>
            <p
              className="card-text"
            >
              As a result of the pandemic, we have set safety measures in place to help keep our customers safe, including numerous sanitising stations and socially distanced seating.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
